packages/components/src/tools-panel/tools-panel-item/hook.ts

```diff
@@ -52,11 +52,10 @@ export function useToolsPanelItem(
 		__experimentalLastVisibleItemClass,
 	} = useToolsPanelContext();
 
-	const hasValueCallback = useCallback( hasValue, [ panelId, hasValue ] );
-	const resetAllFilterCallback = useCallback( resetAllFilter, [
-		panelId,
-		resetAllFilter,
-	] );
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	const hasValueCallback = useCallback( hasValue, [ panelId ] );
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	const resetAllFilterCallback = useCallback( resetAllFilter, [ panelId ] );
```

Comment:
Maybe add some comments to explain why we're disable the eslint rule. 

This is the perfect use-case for the useEvent hook they proposed in React. not sure what happened to that proposal.

Reply:
Wasn't aware of `useEvent`, but we have lots of cases in rich text and elsewhere too where we have to put stuff in a ref to access it in callbacks later.